localstack/utils/sync.py

```diff
@@ -51,7 +51,10 @@ def wait_until(
         return wait_until(fn, next_wait, max_retries, strategy, _retries + 1, _max_wait)
 
 
-def retry(function, retries=3, sleep=1.0, sleep_before=0, **kwargs):
+T = TypeVar("T")
+
+
+def retry(function: Callable[..., T], retries=3, sleep=1.0, sleep_before=0, **kwargs) -> T:
```

Comment:
I got really annoyed my the missing typing here, so I kind of had to do it. I had a more sophisticated version with ParamSpec, but it turns out you cannot use the paramspec kwargs within the same method definition, only within the body of the method.